docs/proposals/service-external-name.md

```diff
@@ -0,0 +1,150 @@
+# Service externalName
+
+Author: Tim Hockin (@thockin), Rodrigo Campos (@rata)
+
+Date: July 2016
+
+Status: Waiting LGTM
+
+# Goal
+
+Allow a service to have a CNAME record in the cluster internal DNS service. For
+example, a `db` service can have a CNAME to `something.rds.aws.amazon.com`
+pointing to an RDS resource.
+
+# Motivation
+
+There were tons of issues about this, I'd try to summarize motivation here. More
+info is on github issues/PRs: #13748, #11838, #13358, #23921
+
+One motivation is to present as cluster services, services that are hosted by
+some provider. Some cloud providers, like AWS, give a hostname (IPs are not
+static) and the user wants to refer using regular kubernetes tools to these
+services. This was asked in bugs, at least for AWS in RedShift, RDS,
+Elasticsearch Service, ELB, etc.
+
+Some others just want an external service, for example "oracle", with dns name
+"oracle-1.testdev.mycompany.com", without having to keep DNS in sync, and just
+want a CNAME.
+
+Another use case is to "integrate" some services to local development. For
+example, I have a search service running in kubernetes in staging, let's say
+`search-1.stating.mycompany.com`. Let's say it's on AWS, so it's behind an ELB
+(which doesn't have a static IP, it has a hostname). I'm building an app that
+consumes `search-1` and I don't want to run it on my local PC (before kubernetes
+I didn't). I can just create a service that has a CNAME to the `search-1`
+endpoint in staging and be happy like I was before.
+
+Also, openshift needs this for "service refs", too:
```

Comment:
On Sun, Jul 17, 2016 at 12:37:18PM -0700, Clayton Coleman wrote:
> > +services. This was asked in bugs, at least for AWS in RedShift, RDS,
> > +Elasticsearch Service, ELB, etc.
> > +
> > +Some others just want an external service, for example "oracle", with dns name
> > +"oracle-1.testdev.mycompany.com", without having to keep DNS in sync, and just
> > +want a CNAME.
> > +
> > +Another use case is to "integrate" some services to local development. For
> > +example, I have a search service running in kubernetes in staging, let's say
> > +`search-1.stating.mycompany.com`. Let's say it's on AWS, so it's behind an ELB
> > +(which doesn't have a static IP, it has a hostname). I'm building an app that
> > +consumes `search-1` and I don't want to run it on my local PC (before kubernetes
> > +I didn't). I can just create a service that has a CNAME to the `search-1`
> > +endpoint in staging and be happy like I was before.
> > +
> > +Also, openshift needs this for "service refs", too:
> 
> To add a bit of color, an OpenShift service ref is really just the three use cases you mentioned above, but in the future we may add a feature to kube that automatically injects service references into namespaces via some sort of "service catalog" (you can probably link to that proposal directly).  The service ref is the natural way to integrate an external service, since it takes advantage of native DNS capabilities already in wide use.

Great, forgot to ask you more details! Thanks, I'll push adding this